backend/Application/AppGenerator.php

```diff
@@ -49,20 +57,32 @@ public function saveDataToJson(DataFactory $factory, array $data): void
 
     public function generateStaticData(): void
     {
+        /** @var Faculty[] $faculties */
+
         $faculties = $this->getDataFromCsv($this->facultyFactory);
         $this->saveDataToJson($this->facultyFactory, $faculties);
-        foreach ($faculties as $faculty) {
-            foreach ($this->subFactories as $subFactory) {
-                OutputWriter::newLineToConsole("Processing {$faculty->getDirectory()}");
-                $subFactory->setDirectory("/faculties/{$faculty->getDirectory()}");
-                $data = $this->getDataFromCsv($subFactory);
-                $this->saveDataToJson($subFactory, $data);
+        try {
+            foreach ($faculties as $faculty) {
+                foreach ($this->subFactories as $subFactory) {
+                    OutputWriter::newLineToConsole("Processing {$faculty->getDirectory()}");
+                    $subFactory->setDirectory("/faculties/{$faculty->getDirectory()}");
+                    $data = $this->getDataFromCsv($subFactory);
+                    $this->saveDataToJson($subFactory, $data);
+                }
+            }
+        } catch (IsMissingValidationException $exception) {
+            OutputWriter::newLineToConsole($exception->getMessage());
+            if ($exception->getFieldName() === "coordinates") {
+                OutputWriter::newLineToConsole("Consider using 'composer fetch' to get missing entries.");
             }
+            die();
         }
     }
 
     public function generateResourceContents(): void
     {
+        /** @var Faculty[] $faculties */
+
```

Comment:
```suggestion
        /** @var Faculty[] $faculties */
```

Reply:
I suspect you want me to remove the new line? Current suggestion doesn't change anything.

Reply:
So, please do it manually